flink-libraries/flink-python/src/main/python/org/apache/flink/python/api/flink/plan/DataSet.py

```diff
@@ -572,6 +572,41 @@ def set_parallelism(self, parallelism):
         self._info.parallelism.value = parallelism
         return self
 
+    def count_elements_per_partition(self):
```

Comment:
please add a docstring for both count_elements_per_pertition and zip_with_index. You should be able to just copy the ones from the Java DataSetUtils class.